packages/spec-core/src/actions/server-test.ts

```diff
@@ -0,0 +1,147 @@
+      }
+      if (mockMethod.response.data) {
+        if (mockMethod.response.data["contentType"] === "application/xml") {
+          if (
+            JSON.stringify(mockMethod.response.data["rawContent"]) !== JSON.stringify(response.data)
+          ) {
+            logger.error(`Response data mismatch for ${this.name} endpoint`);
+            logger.error(
+              `Expected: ${mockMethod.response.data["rawContent"]} - Actual: ${response.data}`,
+            );
+            throw new Error(`Response data mismatch for ${this.name} endpoint`);
+          }
+        } else if (Buffer.isBuffer(mockMethod.response.data)) {
+          if (
+            mockMethod.request.config?.headers &&
+            mockMethod.request.config.headers["accept"] === "application/json"
+          ) {
+            if (response.data.content !== mockMethod.response.data.toString("base64")) {
+              throw new Error(`Response data mismatch for ${this.name} endpoint`);
+            }
+          } else {
+            if (
+              uint8ArrayToString(response.data, "utf-8") !== mockMethod.response.data.toString()
+            ) {
+              throw new Error(`Response data mismatch for ${this.name} endpoint`);
+            }
+          }
+        } else {
+          if (JSON.stringify(mockMethod.response.data) !== JSON.stringify(response.data)) {
+            logger.error(`Response data mismatch for ${this.name} endpoint`);
+            logger.error(`Expected: ${mockMethod.response.data} - Actual: ${response.data}`);
+            throw new Error(`Response data mismatch for ${this.name} endpoint`);
+          }
+        }
+      }
+      if (mockMethod.response.config?.headers) {
+        if (
+          JSON.stringify(mockMethod.response.config.headers) !==
+          JSON.stringify(response.config.headers)
+        ) {
+          throw new Error(`Response headers mismatch for ${this.name} endpoint`);
+        }
+      }
+    }
+  }
+}
+
+export async function serverTest(
+  scenariosPath: string,
+  serverBasePath: string,
+  scenariosConfig: {
+    runSingleScenario: string | undefined;
+    runScenariosFromFile: string | undefined;
+  },
+) {
+  // 1. Get Testcases to run
+  const testCasesToRun: string[] = [];
+  if (scenariosConfig.runSingleScenario) {
+    testCasesToRun.push(scenariosConfig.runSingleScenario);
+  } else if (scenariosConfig.runScenariosFromFile) {
+    const data = fs.readFileSync(path.resolve(scenariosConfig.runScenariosFromFile), "utf8");
+    const lines = data.split("\n");
+    lines.forEach((line) => {
+      testCasesToRun.push(line.trim());
+    });
+  }
+  // 2. Load all the scenarios
+  const scenarios = await loadScenarioMockApis(scenariosPath);
+  // 3. Execute each scenario
+  for (const [name, scenario] of Object.entries(scenarios)) {
+    if (!Array.isArray(scenario.apis)) continue;
+    for (const api of scenario.apis) {
+      if ((api as any as MockApiDefinition).mockMethods === undefined) continue;
```

Comment:
Is putting a kind to discriminate not better than having to cast to any everywhere

Reply:
That's a great idea. Added and modified the code